wandb/sdk/launch/registry/elastic_container_registry.py

```diff
@@ -0,0 +1,126 @@
+"""Implementation of Elastic Container Registry class for wandb launch."""
+import base64
+from typing import Tuple
+
+from wandb.errors import LaunchError
+from wandb.sdk.launch.environment.aws_environment import AwsEnvironment
+from wandb.util import get_module
+
+from .abstract import AbstractRegistry
+
+botocore = get_module(
+    "botocore", required="AWS environment requires botocore to be installed."
+)
+
+
+class ElasticContainerRegistry(AbstractRegistry):
+    """Elastic Container Registry class.
+
+    Attributes:
+        repo_name (str): The name of the repository.
+        environment (AwsEnvironment): The AWS environment.
+        uri (str): The uri of the repository.
+    """
+
+    repo_name: str
+    environment: AwsEnvironment
+    uri: str
+
+    def __init__(self, repo_name: str, environment: AwsEnvironment) -> None:
+        """Initialize the Elastic Container Registry.
+
+        Args:
+            repo_name (str): The name of the repository.
+            environment (AwsEnvironment): The AWS environment.
+
+        Raises:
+            LaunchError: If there is an error verifying the registry.
+        """
+        super().__init__()
+        self.repo_name = repo_name
+        self.environment = environment
+        self.verify()
+
+    def verify(self) -> None:
+        """Verify that the registry is accessible and the configured repo exists.
+
+        Raises:
+            RegistryError: If there is an error verifying the registry.
+        """
+        try:
+            session = self.environment.get_session()
+            client = session.client("ecr")
+            response = client.describe_registry()
+            response = client.describe_repositories(repositoryNames=[self.repo_name])
+            self.uri = response["repositories"][0]["repositoryUri"].split("/")[0]
+
+        except botocore.exceptions.ClientError as e:
+            code = e.response["Error"]["Code"]
+            msg = e.response["Error"]["Message"]
+            # TODO: Log the code and the message here?
```

Comment:
Yep, logging and you can use `traceback.format_exc()` to get a nicely printed message in our logger or you can say 
`raise LaunchError(...) from e`

Reply:
I haven't changed this yet- it's currently raising an exception with the error code and message which I think is par this part of the code base. I would like to do a pass after getting the larger changes in to do standardize and improve the logging from these components